File: src/Http.API/ClientApp/src/app/share/admin/services/entity-member.service.ts
import { Injectable } from '@angular/core';
import { BaseService } from './base.service';
import { Observable } from 'rxjs';
import { EntityMemberFilterDto } from '../models/entity-member/entity-member-filter-dto.model';
import { EntityMemberAddDto } from '../models/entity-member/entity-member-add-dto.model';
import { EntityMemberUpdateDto } from '../models/entity-member/entity-member-update-dto.model';
import { EntityMemberItemDtoPageList } from '../models/entity-member/entity-member-item-dto-page-list.model';
import { EntityMember } from '../models/entity-member/entity-member.model';

/**
 * 实体属性
 */
@Injectable({ providedIn: 'root' })
export class EntityMemberService extends BaseService {
  /**
   * 筛选
   * @param data EntityMemberFilterDto
   */
  filter(data: EntityMemberFilterDto): Observable<EntityMemberItemDtoPageList> {
    const url = `/api/admin/EntityMember/filter`;
    return this.request<EntityMemberItemDtoPageList>('post', url, data);
  }

  /**
   * 新增
   * @param data EntityMemberAddDto
   */
  add(data: EntityMemberAddDto): Observable<EntityMember> {
    const url = `/api/admin/EntityMember`;
    return this.request<EntityMember>('post', url, data);
  }

  /**
   * 更新
   * @param id 
   * @param data EntityMemberUpdateDto
   */
  update(id: string, data: EntityMemberUpdateDto): Observable<EntityMember> {
    const url = `/api/admin/EntityMember/${id}`;
    return this.request<EntityMember>('put', url, data);
  }

  /**
   * 详情
   * @param id 
   */
  getDetail(id: string): Observable<EntityMember> {
    const url = `/api/admin/EntityMember/${id}`;
    return this.request<EntityMember>('get', url);
  }

  /**
   * ⚠删除
   * @param id 
   */
  delete(id: string): Observable<EntityMember> {
    const url = `/api/admin/EntityMember/${id}`;
    return this.request<EntityMember>('delete', url);
  }

}

File: src/Sdks/Dusi.Manage.Client/Services/EntityMemberService.cs
using Share.Models.EntityMemberDtos;
namespace Dusi.Manage.Client.Services;
/// <summary>
/// 实体属性
/// </summary>
public class EntityMemberService : BaseService
{
    public EntityMemberService(HttpClient httpClient) : base(httpClient)
    {

    }
    /// <summary>
    /// 筛选
    /// </summary>
    /// <param name="data">EntityMemberFilterDto</param>
    /// <returns></returns>
    public async Task<PageList<EntityMemberItemDto>?> FilterAsync(EntityMemberFilterDto data)
    {
        string url = $"/api/admin/EntityMember/filter";
        return await PostJsonAsync<PageList<EntityMemberItemDto>?>(url, data);
    }

    /// <summary>
    /// 新增
    /// </summary>
    /// <param name="data">EntityMemberAddDto</param>
    /// <returns></returns>
    public async Task<EntityMember?> AddAsync(EntityMemberAddDto data)
    {
        string url = $"/api/admin/EntityMember";
        return await PostJsonAsync<EntityMember?>(url, data);
    }

    /// <summary>
    /// 更新
    /// </summary>
    /// <param name="id"> </param>
    /// <param name="data">EntityMemberUpdateDto</param>
    /// <returns></returns>
    public async Task<EntityMember?> UpdateAsync(string id, EntityMemberUpdateDto data)
    {
        string url = $"/api/admin/EntityMember/{id}";
        return await PutJsonAsync<EntityMember?>(url, data);
    }

    /// <summary>
    /// 详情
    /// </summary>
    /// <param name="id"> </param>
    /// <returns></returns>
    public async Task<EntityMember?> GetDetailAsync(string id)
    {
        string url = $"/api/admin/EntityMember/{id}";
        return await GetJsonAsync<EntityMember?>(url);
    }

    /// <summary>
    /// ⚠删除
    /// </summary>
    /// <param name="id"> </param>
    /// <returns></returns>
    public async Task<EntityMember?> DeleteAsync(string id)
    {
        string url = $"/api/admin/EntityMember/{id}";
        return await DeleteJsonAsync<EntityMember?>(url);
    }

}